### PR TITLE
Fix pre-round lobby map vote filter

### LINKED ISSUE #73816: Map votes don't filter if you start them in the pre-round lobby
Oversight, can't count active players before the round starts.

## Reproduction:

Hold a map vote.

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request

Fixes https://github.com/tgstation/tgstation/issues/73816

## Changelog

:cl: LT3
fix: Map filter works properly in the pre-round lobby
fix: Map vote doesn't log 'not enough players' if the vote starts in the pre-round lobby
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/mapping.dm [hunk 1]
```diff
@@ -531,10 +531,11 @@ GLOBAL_LIST_EMPTY(the_station_areas)
 		message_admins("Failed to set new map with next_map.json for [change_to.map_name]! Using default as backup!")
 		return
 
-	if (change_to.config_min_users > 0 && GLOB.clients.len < change_to.config_min_users)
+	var/filter_threshold = get_active_player_count(alive_check = FALSE, afk_check = TRUE, human_check = FALSE)
+	if (change_to.config_min_users > 0 && filter_threshold != 0 && filter_threshold < change_to.config_min_users)
```
Comment (Jacquerel) [inline code/controllers/subsystem/mapping.dm:535]: Is the `filter_threshold != 0` check here necessary? What circumstances would have it be 0, and do we not want to log that we are entering delta with 0 active players?

**Review by Jacquerel — CHANGES_REQUESTED** [review]
Mostly nitpicking here but I want to be sure that some decisions were intentional and not just the most convenient thing to do.

File: code/controllers/subsystem/mapping.dm (hunk 1 above)
Comment (lessthnthree) [reply code/controllers/subsystem/mapping.dm:535]: pre-round lobby it is 0.

File: code/datums/votes/map_vote.dm [hunk 2]
```diff
@@ -67,21 +67,24 @@
...
+/// Note that this can result in zero remaining choices for our vote, which is not ideal (but ultimately okay).
 /// Argument should_key_choices is TRUE, pass as FALSE in a context where choices are already keyed in a list.
 /datum/vote/map_vote/proc/check_population(should_key_choices = TRUE)
+	var/filter_threshold = get_active_player_count(alive_check = FALSE, afk_check = TRUE, human_check = FALSE)
 	if(should_key_choices)
 		for(var/key in default_choices)
 			choices[key] = 0
 
-	var/active_players = get_active_player_count(alive_check = FALSE, afk_check = TRUE, human_check = FALSE)
+	if(!filter_threshold) // Pre-round lobby won't have an active player count yet, so fallback.
```
[the remark this answers is not in the document]
Comment (lessthnthree) [reply code/datums/votes/map_vote.dm:79]: Yeah, this is a better way of doing it. Didn't know we had a HasRoundStarted proc.

File: code/controllers/subsystem/mapping.dm [hunk 3]
```diff
@@ -531,10 +531,11 @@ GLOBAL_LIST_EMPTY(the_station_areas)
...
 		message_admins("[change_to.map_name] was chosen for the next map, despite there being less current players than its set minimum population range!")
 		log_game("[change_to.map_name] was chosen for the next map, despite there being less current players than its set minimum population range!")
-	if (change_to.config_max_users > 0 && GLOB.clients.len > change_to.config_max_users)
+	if (change_to.config_max_users > 0 && filter_threshold != 0 && filter_threshold > change_to.config_max_users)
```
[the remark this answers is not in the document]
Comment (lessthnthree) [reply code/controllers/subsystem/mapping.dm:538]: We don't confirm it here, but yeah it's only needed for the max check

File: code/datums/votes/map_vote.dm [hunk 4]
```diff
@@ -67,21 +67,24 @@
 	message = initial(message)
 	return TRUE
 
-/// Before we create a vote, remove all maps from our choices that are outside of our population range. Note that this can result in zero remaining choices for our vote, which is not ideal (but ultimately okay).
+/// Before we create a vote, remove all maps from our choices that are outside of our population range.
+/// Note that this can result in zero remaining choices for our vote, which is not ideal (but ultimately okay).
 /// Argument should_key_choices is TRUE, pass as FALSE in a context where choices are already keyed in a list.
 /datum/vote/map_vote/proc/check_population(should_key_choices = TRUE)
+	var/filter_threshold = get_active_player_count(alive_check = FALSE, afk_check = TRUE, human_check = FALSE)
```
[the remark this answers is not in the document]
Comment (lessthnthree) [reply code/datums/votes/map_vote.dm]: Moved it back, I thought style was all new vars below the start of the proc

File: code/controllers/subsystem/mapping.dm (hunk 1 above)
Comment (Jacquerel) [reply code/controllers/subsystem/mapping.dm:535]: I think I have now forgotten why I asked that question so it's probably fine.

**Pull request merged**